### PR TITLE
Added conditionals for other combinations of fields.

### PR DESCRIPTION
**Pull request by mdyoung3**
# READY FOR REVIEW

# Summary
- TL;DR - what's this PR for?
- Missing commas on the Opportunity grid  cards for the fields that appear together. 
- Ticket: https://stanfordits.atlassian.net/browse/CS-112

# Review By (Date)
- Monday December 11

# Criticality
- 5
- 
# Urgency
- Normal

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Rebuild Cache 
3. Go to Opportunities page and make sure none of the fields abut each other and are missing commas. 

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory?

## Front End Validation
- [ ] Design is approved by @ user?
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

**Review by joegl — CHANGES_REQUESTED** [review]
I don't think this logic catches everything. 

You wouldn't want to add a comma to service_theme or time_commitment if the next two data points (program or type_list) are empty (then there'd just be a hanging comma).

The problem with the existing logic is it only adds a comma after the service_theme if both a time_commitment and service_theme exist, because the condition for adding a comma to the service_theme is nested. The display issue occurring on the site is there are Opportunities with a service_theme, but without a time_commitment (so the nested condition which adds a comma after the service_theme doesn't ever run). If we un-nest the condition that adds a comma after the service_theme, it should resolve the on-site issue. 

It still wouldn't account for the edge case when there is a time_commitment but not a service_theme, but I don't know if that is a plausible scenario. You could also re-write the entire logic as you see fit. The important parts are:

* If there is both a time_commitment AND a service_theme it should output as "time_commitment **in** service_theme" (these two don't get comma separated).
* Anything else is comma separated
